lib/enmeshed/relationship.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Enmeshed
+  class Relationship
+    STATUS_GROUP_SYNONYMS = JSON.parse(File.read('lib/enmeshed/status_group_synonyms.json'))
+
+    def initialize(relationship_json)
+      @json = relationship_json
+      @relationship_changes = relationship_json[:changes] || []
+    end
+
+    def peer
+      @json[:peer]
+    end
+
+    def userdata
+      @userdata ||= parse_userdata
+    end
+
+    def template_expiration
+      DateTime.parse(@json[:template][:expiresAt])
```

Comment:
After applying my suggestion from https://github.com/openHPI/codeharbor/pull/1326/files#r1635394823, we can also improve this method. It operations on the `Relationship` object, besides queuing data from the `RelationshipTemplate`. Hence, it would be ideal to delegate it to the `template`, too (as introduced in my previous comment). Then, we mainly need another constructor for the template, which could then accept the truncated reference only or the whole template:

*Pseudo-Code*

```rb
module Enmeshed
  class RelationshipTemplate
    attr_reader :expires_at

    def initialize(truncated_reference, content)
      @truncated_reference = truncated_reference
      if content
        @expires_at = DateTime.parse(content[:expiresAt])
        # Potentially parse and set more values based on the existing request
      end
    end

    def self.from_request(content)
      self.new(content[:truncatedReference], content)
    end
  end
end
```

Reply:
When we move the template generation itself (see https://github.com/openHPI/codeharbor/pull/1326#discussion_r1635399263), we might also be able to combine that with the initialization here. Potentially, this could work similar to the `identity_attribute`: Either, we just return the content that the `RelationshipTemplate` was initialized with (in case of an existing `RelationshipTemplate`) or we draft a new one (the current `Connector.relationship_template` method). Then, the attributes we access `expires_at` would be a simple reader method, returning the value:

```rb
def expires_at
  @expires_at ||= DateTime.parse(content[:expiresAt])
end

Reply:
I inlined `template_expiration`. Do you think we should still move this into `RelationshipTemplate`?

Reply:
Yes, I'd think so. I made a few suggestions on how that could look like in dedicated commits (so that they are easier to understand), which we should squash before merging.

Reply:
Looks good :+1: